content.js

```diff
@@ -29,34 +47,58 @@ function toggleFiles() {
 }
 
 function addToggleBtn() {
-	const toggleBtn = createHtml('<a class="hide-files-btn btn btn-sm">Show dotfiles</a>');
+	const toggleBtn = createHtml(`<a class="hide-files-btn btn btn-sm">${label()}</a>`);
 	const btnContainer = document.querySelector('.file-navigation .right');
 
 	if (document.querySelector('.hide-files-btn')) {
+		addToggleBtnEvents();
 		return;
 	}
 
 	if (btnContainer) {
 		// insert after
 		btnContainer.insertBefore(toggleBtn, btnContainer.children[0]);
+		addToggleBtnEvents();
+	}
+}
 
-		const btn = document.querySelector('.hide-files-btn');
+function addToggleBtnEvents() {
+	const btn = document.querySelector('.hide-files-btn');
 
+	if (btn) {
 		btn.addEventListener('click', () => {
-			isHidden = !isHidden;
-			btn.textContent = isHidden ? 'Show dotfiles' : 'Hide dotfiles';
+			toggleOn = !toggleOn;
+			btn.textContent = label();
 			toggleFiles();
 		});
 	}
 }
 
+function label() {
+	return toggleOn ? 'Show dotfiles' : 'Hide dotfiles';
+}
+
 function trigger() {
 	addToggleBtn();
 	toggleFiles();
 }
 
 document.addEventListener('DOMContentLoaded', () => {
-	trigger();
-
 	new MutationObserver(trigger).observe(document.querySelector('#js-repo-pjax-container'), {childList: true});
+
+	window.chrome.storage.sync.get({
+		visibility: 'hidden',
+		ignoreRegEx: ''
```

Comment:
The funny side effect of allowing everything to pass through the regex and using a default value of:
`^\.|^license|^appveyor\.yml` is that now everything *except* the files that match the pattern are hidden, because right now the `ignoreRegExp` input field in the options page is acting like an "ignore these files [i.e, dont hide these files]" so do I have things backwards? Should the `ignoreRegExp` input field in the options page be setting the names of files to actually hide?